crda does not build with libressl

## Issue description
When globally overriding `openssl` with `libressl`, I now see that `crda` fails to build. It's weird because I have been doing this for a long time and never had a problem with it.

I know libressl was updated to the 2.8 series recently, but I tried rebuilding my system with all of 2.7, 2.8 and 2.9 and get this with them all. I am confused!

Now, who, out of `crda`, `m2crypto` and `libressl` is at fault here?

The build output (which I have updated the crda package to 4.14, but this makes no difference) is:
```
$ nixos-rebuild build 
building Nix...
building the system configuration...
these derivations will be built:
  /nix/store/gmyjjsk68pw3ff3zjglccy3isyv1nlvl-crda-4.14.drv
  /nix/store/mdxy11c47i51kcaahspyzh652s2anxd9-unit-systemd-udevd.service.drv
  /nix/store/7w8sbqjgmwgl61py3qxcbzd4frpp4l5j-system-units.drv
  /nix/store/y0r9c149z2pj26yvyjxfivhywq55681f-udev-rules.drv
  /nix/store/y2277xdpwgxk1nsf4g87dka96cn4j85p-hwdb.bin.drv
  /nix/store/djhlkn3kh6pbjjbxh8hf2mb5jw1gac6a-etc.drv
  /nix/store/gh398fzd62l4kn58jcsx5ym30p27vzwd-nixos-system-braid-19.03.git.1a7d28e.drv
building '/nix/store/gmyjjsk68pw3ff3zjglccy3isyv1nlvl-crda-4.14.drv'...
unpacking sources
unpacking source archive /nix/store/0xavp3ccpz3h7plvdrxb88xr9cq2hbdw-crda-4.14.tar.gz
source root is crda-4.14
setting SOURCE_DATE_EPOCH to timestamp 1515178391 of file crda-4.14/utils/key2pub.py
patching sources
patching script interpreter paths in utils/
utils/key2pub.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/qxzffzmkzm88407y7khwxxx5wbdsys54-python-2.7.15/bin/python"
configuring
no configure script, doing nothing
building
build flags: -j4 -l4 SHELL=/nix/store/can00lfiynqkbsdkkmgp6qg8p8w92cxa-bash-4.4-p23/bin/bash PREFIX=\$\(out\) SBINDIR=\$\(out\)/bin/ UDEV_RULE_DIR=\$\(out\)/lib/udev/rules.d/ REG_BIN=/nix/store/qxp5pdxcdllwd452y3h6sc3jfm68hs15-wireless-regdb-2018.10.24/lib/crda/regulatory.bin all_noverify
  GEN  keys-gcrypt.c
  Trusted pubkeys: pubkeys/sforshee.key.pub.pem pubkeys/linville.key.pub.pem
ERROR: Failed to import the "M2Crypto" module: /nix/store/nkaxq5j9392jfp5x6xvpn8hd6i17vg0b-python2.7-M2Crypto-0.31.0/lib/python2.7/site-packages/M2Crypto/_m2crypto.so: undefined symbol: BIO_get_init
Please install the "M2Crypto" Python module.
On Debian GNU/Linux the package is called "python-m2crypto".
make: *** [Makefile:115: keys-gcrypt.c] Error 1
builder for '/nix/store/gmyjjsk68pw3ff3zjglccy3isyv1nlvl-crda-4.14.drv' failed with exit code 2
cannot build derivation '/nix/store/y2277xdpwgxk1nsf4g87dka96cn4j85p-hwdb.bin.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/y0r9c149z2pj26yvyjxfivhywq55681f-udev-rules.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/mdxy11c47i51kcaahspyzh652s2anxd9-unit-systemd-udevd.service.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/7w8sbqjgmwgl61py3qxcbzd4frpp4l5j-system-units.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/djhlkn3kh6pbjjbxh8hf2mb5jw1gac6a-etc.drv': 3 dependencies couldn't be built
cannot build derivation '/nix/store/gh398fzd62l4kn58jcsx5ym30p27vzwd-nixos-system-braid-19.03.git.1a7d28e.drv': 1 dependencies couldn't be built
error: build of '/nix/store/gh398fzd62l4kn58jcsx5ym30p27vzwd-nixos-system-braid-19.03.git.1a7d28e.drv' failed
```

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.
```
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.13, NixOS, 19.03.git.1a7d28e (Koi)`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.1.3`
 - channels(root): `"nixos-18.09pre133640.ea145b68a01, nixpkgs-18.09pre133932.ee28e35ba37"`
 - nixpkgs: `/nix/nixpkgs`
```

--- a/pkgs/development/python-modules/m2crypto/default.nix
+++ b/pkgs/development/python-modules/m2crypto/default.nix
@@ -1,4 +1,5 @@
 { stdenv
+, fetchpatch
 , buildPythonPackage
 , fetchPypi
 , swig2
@@ -16,6 +17,14 @@ buildPythonPackage rec {
     sha256 = "a1b2751cdadc6afac3df8a5799676b7b7c67a6ad144bb62d38563062e7cd3fc6";
   };
 
+  patches = [
+    (fetchpatch {
+      url = "https://github.com/void-linux/void-packages/raw/7946d12eb3d815e5ecd4578f1a6133d948694370/srcpkgs/python-M2Crypto/patches/libressl.patch";
+      sha256 = "0z5qnkndg6ma5f5qqrid5m95i9kybsr000v3fdy1ab562kf65a27";
+    })
+  ];
+  patchFlags = "-p0";
+
   buildInputs = [ swig2 openssl ];
 
   propagatedBuildInputs = [ typing ];
@@ -30,6 +39,7 @@ buildPythonPackage rec {
     description = "A Python crypto and SSL toolkit";
     homepage = http://chandlerproject.org/Projects/MeTooCrypto;
     license = licenses.mit;
+    maintainers = with maintainers; [ andrew-d ];
   };
 
 }

